python/app/dialog.py

```diff
@@ -0,0 +1,793 @@
+            self.ui.current_user.setToolTip("%s's Home" % sg_data["firstname"])
+
+
+    def _refresh_details_thumbnail(self):        
+        self.ui.details_thumb.setPixmap(self._details_model.get_pixmap())
+
+    def _refresh_details(self):
+        
+        formatter = self._current_location.sg_formatter        
+        sg_data = self._details_model.get_sg_data()
+        
+        # set up the thumbnail
+        self.ui.details_thumb.set_shotgun_data(sg_data)
+        
+        # populate the text with data
+        if sg_data:
+            (header, body) = formatter.format_entity_details(sg_data) 
+            self.ui.details_text_header.setText(header)
+            self.ui.details_text_header.setToolTip(header)
+            self.ui.details_text_middle.setText(body)
+            
+        else:
+            self.ui.details_text_header.setText("")
+            self.ui.details_text_header.setToolTip("")
+            self.ui.details_text_middle.setText("")
+            
+        # load actions
+        actions = self._action_manager.get_actions(sg_data, 
+                                                   self._action_manager.UI_AREA_DETAILS)
+        self._actions = actions
+        for a in self._actions:
+            self._menu.addAction(a)
+            
+            
+            
+    ###################################################################################################
+    # UI callbacks
+    def _on_entity_clicked(self, model_index):
+        """
+        Someone clicked an entity
+        """
+        sg_item = shotgun_model.get_sg_data(model_index)
+        sg_location = ShotgunLocation(sg_item["type"], sg_item["id"])
+        self._navigate_to(sg_location)
+
+    def _navigate_to_entity(self, entity_type, entity_id):
+        """
+        Navigate to a particular entity 
+        """
+        sg_location = ShotgunLocation(entity_type, entity_id)
+        if sg_location.sg_formatter.should_open_in_shotgun_web:
+            sg_url = sg_location.get_external_url()
+            QtGui.QDesktopServices.openUrl(QtCore.QUrl(sg_url))            
+        else:
+            self._navigate_to(sg_location)
+        
+    def _playback_version(self, version_data):
+        """
+        Given version data, play back a version
+        
+        :param version_data: A version dictionary containing version data
+        """
+        url = ShotgunFormatter.get_playback_url(version_data)
+        if url:
+            QtGui.QDesktopServices.openUrl(QtCore.QUrl(url))
+        else:
+            self._app.log_warning("Cannot play back version %s - "
+                                  "no playback url defined." % version_data["id"])
+        
+        
+
+    def _on_link_clicked(self, url):
+        """
+        When someone clicks a url
+        """
+        if url is None:
+            return
+        
+        if url.startswith("http"):
+            QtGui.QDesktopServices.openUrl(QtCore.QUrl(url))
+            
+        else:
```

Comment:
hmm, yes that's a good point... I'll flip the logic around so that it looks for the internal form and for everything else launches the OS integration...